libs/domain/product/media/src/media.component.ts

```diff
@@ -32,14 +32,34 @@ export class ProductMediaComponent extends ProductMixin(
       containerSize
     );
 
+    if (sources[0]?.url) {
```

Comment:
```
if (sources[0]?.url)  {return html``};

if (this.isVideo(sources[0]?.url)) {
        return this.renderVideo(sources[0]?.url);
}

return this.renderImage(sources[0]?.url, this.getSrcSet(sources));
```
 
And another thing. I noticed that you call service inside render and try to get the source.
```
const sources = this.imageService.resolveSources(
      productMedia,
      containerSize
);
```

It will call only during re-render, and if for some reason service resolves the source after last render it doesn't affect on template, because this is not reactive

Reply:
Good point, i'm changing to update lifecycle